api/v1beta1/bucket_types.go

```diff
@@ -30,6 +28,18 @@ const (
 	BucketKind = "Bucket"
 )
 
+const (
+	GenericBucketProvider string = "generic"
+	AmazonBucketProvider  string = "aws"
+)
+
+const (
+	// DownloadFailedCondition indicates a transient or persistent download failure. If True, observations on the
+	// upstream Source revision are not possible, and the Artifact available for the Source may be outdated.
```

Comment:
Note to self: with current code changes the observation of a revision may still be possible, while getting the object file at a later moment may fail.